app/routes/products.$categorySlug/use-product-filters.ts

```diff
@@ -0,0 +1,73 @@
+import { useCallback, useMemo } from 'react';
+import { useSearchParams } from '@remix-run/react';
+import { IProductFilters, ProductFilter } from '~/api/types';
+
+export function useProductFilters() {
```

Comment:
If this can be used in other templates with no changes, let's move this to the `api` folder.